go.mod

```diff
@@ -30,6 +30,7 @@ require (
 	github.com/konsorten/go-windows-terminal-sequences v1.0.2 // indirect
 	github.com/mailru/easyjson v0.0.0-20190403194419-1ea4449da983 // indirect
 	github.com/mitchellh/go-homedir v1.1.0 // indirect
+	github.com/mitchellh/mapstructure v1.3.0
```

Comment:
It doesn't? this basically is used for mapping a `map[string]interface{}` to a custom type, e.g. `estypes.Index`

Reply:
this is what i was wondering too... but i think to do that we'd need to know the custom type at the time of the curl request so that we can use that for unmarshalling into.

Reply:
This is basically a hint that we need to rework the ES client underneath to provide static types, because working with nested result `map[string]inteface{}` needs a lot of castings. This is an easy way to define a type and unmarshall into it.

Reply:
I replaced the lib with unmarshalling the raw body response into the appropriate type.